sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -173,7 +171,20 @@ case class ScriptTransformation(
           }
           outputStream.close()
         }
-      }).start()
+      }, "Thread-ScriptTransformation-Feed").start()
+
+      // Consume the error stream from the pipeline, otherwise it will be blocked if
+      // the pipeline is full.
+      new Thread(new Runnable() {
```

Comment:
Instead of writing another thread class, can you use `RedirectThread`, which is used for the same purpose elsewhere, to dump the data? 
Also, it's slow to read a byte at a time here.